src/SprykerEco/Yves/Heidelpay/Theme/default/components/molecules/direct-debit-new-registration/direct-debit-new-registration.ts

```diff
@@ -0,0 +1,75 @@
+import Component from 'ShopUi/models/component';
+
+export default class DirectDebitNewRegistration extends Component {
+
+    protected inputs: HTMLInputElement[];
+    protected paymentForm: HTMLFormElement;
+    protected paymentMethodToggler: HTMLInputElement;
+    protected paymentFormTogglers: HTMLInputElement[];
+    protected directDebitPaymentToggler: HTMLInputElement;
+    protected paymentName: string = 'heidelpayDirectDebit';
+    protected paymentOptionsTogglers: HTMLInputElement[];
+
+    protected readyCallback(): void {}
+
+    protected init(): void {
+        this.paymentForm = <HTMLFormElement>document.getElementById(this.parentFormId);
+        this.inputs = Array.from(this.getElementsByTagName('input'));
+        this.paymentOptionsTogglers = <HTMLInputElement[]>Array.from(this.paymentForm.querySelectorAll(`input[name="${this.paymentOptionInputName}"`));
+        this.directDebitPaymentToggler = <HTMLInputElement>this.paymentOptionsTogglers.find(toggler => toggler.value === this.name);
+        this.paymentFormTogglers = <HTMLInputElement[]>Array.from(this.paymentForm.querySelectorAll(this.paymentTogglerSelector));
+        this.paymentMethodToggler = <HTMLInputElement>this.paymentFormTogglers.find(toggler => toggler.value === this.paymentName);
+        this.mapEvents();
+    }
+
+    protected mapEvents(): void {
+        this.paymentForm.addEventListener('submit', this.onSubmit.bind(this));
+    }
+
+    protected onSubmit(event: Event): void {
+        if (!this.isDirectDebitNewRegistrationActive()) return;
+        event.preventDefault();
+        this.changeFormAction();
+        this.disableFormFields();
+        this.paymentForm.submit();
+
+    }
+
+    protected disableFormFields(): void {
+        const elements = <HTMLInputElement[]>Array.from(this.paymentForm.elements);
+        elements.forEach(this.checkInputsToDisable.bind(this));
+    }
+
+    protected checkInputsToDisable(element: HTMLInputElement): void {
+        this.inputs.forEach(input => {
+            element.disabled = true;
```

Comment:
Why do you mix `inputs` and `elements`. 
You can first at all disable all inputs and in another method enable elements...

Reply:
fixed